### ISSUE TITLE
Optimization, multiple GPU

### ISSUE THREAD

**Issue by sergman2** [matterport/Mask_RCNN#312]
I am trying to optimize the current detection model which takes around 400 ms per frame (1080i). The goal is to reach realtime detection using multiple GPU/ Servers.  I work with specific images in which I know approximate area where the object is. So my thought was to mask or even cut the whole image and feed just an area where I am interested to find the object. Unfortunately this did not improve detection time. My second thought was to tell the model to find for specific class and not all the classes it was trained for so it would speed up detection time. But I could not find such an option. 

My second issue is that I was not able to utilize second GPU. In the code there are lines:
    if args.command == "train":
        config = SoccerConfig()
    else:
        class InferenceConfig(SoccerConfig):
            # Set batch size to 1 since we'll be running inference on
            # one image at a time. Batch size = GPU_COUNT * IMAGES_PER_GPU
            GPU_COUNT = 1
            IMAGES_PER_GPU = 1
            DETECTION_MIN_CONFIDENCE = 0

which override GPU_COUNT for detection. I tried to set it to 2 and submit two images to .detect function but I receive Tensorflow error:
Invalid argument: Input to reshape is a tensor with 600 values, but the requested shape has 1200

Is it possible to set model configuration to a better performance even if I have to pay with accuracy?

My OS is Windows, CUDA8, two 1080i,  and config is as follows:

Configurations:
BACKBONE_SHAPES                [[256 256]
 [128 128]
 [ 64  64]
 [ 32  32]
 [ 16  16]
 [  8   8]
 [  4   4]]
BACKBONE_STRIDES               [4, 8, 16, 32, 64, 128, 256]
BATCH_SIZE                     1
BBOX_STD_DEV                   [ 0.1  0.1  0.2  0.2]
DETECTION_MAX_INSTANCES        100
DETECTION_MIN_CONFIDENCE       0
DETECTION_NMS_THRESHOLD        0.3
GPU_COUNT                      1
IMAGES_PER_GPU                 1
IMAGE_MAX_DIM                  1024
IMAGE_MIN_DIM                  800
IMAGE_PADDING                  True
IMAGE_SHAPE                    [1024 1024    3]
LEARNING_MOMENTUM              0.9
LEARNING_RATE                  0.001
MASK_POOL_SIZE                 14
MASK_SHAPE                     [28, 28]
MAX_GT_INSTANCES               100
MEAN_PIXEL                     [ 123.7  116.8  103.9]
MINI_MASK_SHAPE                (56, 56)
NAME                           test
NUM_CLASSES                    30
POOL_SIZE                      7
POST_NMS_ROIS_INFERENCE        1000
POST_NMS_ROIS_TRAINING         2000
ROI_POSITIVE_RATIO             0.33
RPN_ANCHOR_RATIOS              [0.5, 1, 2]
RPN_ANCHOR_SCALES              (16, 32, 64, 128, 256, 512, 1024)
RPN_ANCHOR_STRIDE              1
RPN_BBOX_STD_DEV               [ 0.1  0.1  0.2  0.2]
RPN_NMS_THRESHOLD              0.7
RPN_TRAIN_ANCHORS_PER_IMAGE    256
STEPS_PER_EPOCH                1000
TRAIN_ROIS_PER_IMAGE           200
USE_MINI_MASK                  True
USE_RPN_ROIS                   True
VALIDATION_STEPS               50
WEIGHT_DECAY                   0.0001

**Comment by keineahnung2345** [comment]
I have met the problem of `Invalid argument: Input to reshape is a tensor with 600 values, but the requested shape has 1200`. And I guess it's because you can't do 2 GPU inference using a model that is previously trained on 1 GPU.

**Comment by chAwater** [comment]
@sergman2 @keineahnung2345 

Please check this PR #1082, it works for me.
